src/client/Microsoft.Identity.Client/ManagedIdentity/V2/ImdsV2ManagedIdentitySource.cs

```diff
@@ -199,10 +203,60 @@ private async Task<CertificateRequestResponse> ExecuteCertificateRequestAsync(st
                 { OAuth2Header.XMsCorrelationId, _requestContext.CorrelationId.ToString() }
             };
 
+            string attestationJwt = string.Empty;
+
+            // Normalize endpoint
+            string normalizedEndpoint = NormalizeAttestationEndpoint(attestationEndpoint, _requestContext.Logger);
+            if (string.IsNullOrEmpty(normalizedEndpoint))
+            {
+                throw MsalServiceExceptionFactory.CreateManagedIdentityException(
+                    "attestation_endpoint_invalid",
+                    $"[ImdsV2] Attestation endpoint '{attestationEndpoint}' is invalid or unsupported.",
+                    null, ManagedIdentitySource.ImdsV2, null);
+            }
+
+            if (managedIdentityKeyInfo.Type == ManagedIdentityKeyType.KeyGuard)
+            {
+                // 1) Resolve provider (must be installed via .WithMtlsProofOfPossession(), see #5483)
+                var provider = _requestContext.AttestationTokenProvider
+                    ?? throw MsalServiceExceptionFactory.CreateManagedIdentityException(
+                        "attestation_provider_missing",
+                        "[ImdsV2] KeyGuard key requires attestation, but no provider is registered.",
```

Comment:
This is not a user friendly exception, but I also don't think the user can trigger this flow without installing the provider package.

Reply:
removed